### PR TITLE
feat(useTemporalNow): add useTemporalNow

### PR DESCRIPTION
**Pull request by JonathanSchndr**
### Description

Issue: #4792 

This PR introduces `useTemporal`, a new composable that provides a reactive wrapper around the modern Temporal API (TC39 proposal) using the `@js-temporal/polyfill` package. This implementation addresses the limitations of the legacy `Date` object by providing:

- 🌍 Robust timezone handling with conversion utilities
- 📅 Support for multiple calendar systems (Gregorian, Islamic, Hebrew, etc.)
- ⚡ Configurable reactive updates
- 🎯 Comprehensive TypeScript support
- 🔧 Rich set of date/time manipulation functions

The implementation includes:
- Full test coverage (index.test.ts)
- Comprehensive documentation (index.md)
- Interactive demo component (demo.vue)
- Two main exports:
  - `useTemporal()`: Reactive composable
  - `createTemporal()`: Static utility function

I have:
- [x] Read the Contributing Guidelines
- [x] Read the Pull Request Guidelines
- [x] Verified no similar PR exists
- [x] Included comprehensive tests
- [x] Added detailed documentation
- [x] Created a demo component

### Additional context

This implementation is particularly focused on:
1. **Type safety**: Full TypeScript support with proper Temporal types
2. **Flexibility**: Support for different timezones and calendar systems
3. **Performance**: Efficient updates using shallowRef
4. **Testing**: Comprehensive test coverage including edge cases
5. **Documentation**: Detailed API documentation with examples

The demo component showcases real-world usage scenarios including:
- World clock functionality
- Calendar system conversion
- Duration-based calculations
- Formatting options

### CODE REVIEW

**Comment by JonathanSchndr** [comment]
I would like to discuss again whether https://github.com/fullcalendar/temporal-polyfill is not the better polyfill? What is your opinion?

Screenshot 2025-06-04 at 08 48 15

**Review by antfu — COMMENTED** [review]
We are on hold for introducing new functions, and also that Temporal is still in Stage 3, https://github.com/tc39/proposal-temporal. I think we'd better leave it to community until the propose is land.

**Comment by JonathanSchndr** [comment]
> We are on hold for introducing new functions, and also that Temporal is still in Stage 3, https://github.com/tc39/proposal-temporal. I think we'd better leave it to community until the propose is land.

My reason was that Firefox is now the first browser to support this natively and I have the hope that the others will soon follow suit. Therefore, my idea was to support this transition with the polyfill. But let’s wait until it has a better native coverage, then the polyfill would be removed and the PR would revise it. What do you think of the procedure? I think everyone hates new date 😅

**Comment by antfu** [comment]
Stage 3 is not stable enough. Once the proposal is landed, I think we can have it in the core package and without the polyfill (polyfill should be something for users to install themselves if needed)

**Comment by JonathanSchndr** [comment]
Great, then I would wait for it and adjust the PR as soon as it has increased the coverage and stability. You are also welcome to ping me if you think it is the right time. Thank you 😊👍

**Comment by 9romise** [comment]
Now that [Temporal](https://github.com/tc39/proposal-temporal) has reached [Stage 4](https://github.com/tc39/proposals/blob/main/finished-proposals.md), I think it's time to move this PR forward!

**Review by antfu — CHANGES_REQUESTED** [review]
I would love to have this landed. Thank you

But as this is something going to be standard, I don't think we should deps on the ployfill, instead we can write in the docs to for users to isntall the polyfill on their own.

**Review by antfu — APPROVED** [review]
I have pushed some changes:
- Decoupled the polyfill, allow passing the Termporal object
- removed `createTemporal` for now as I am not sure how much value it provides at the current stage, we can keep discussing and bring it back in another PR if we found it useful.
- renamed the function to `useTemporalNow` to be more clear. Later we might think to have a more general `useTemporal` but right now it seems the composable is focusing on the current time.

**Comment by JonathanSchndr** [comment]
Thank you 🙏 if I can still support you somewhere, please let me know ☺️

**Review by 9romise — APPROVED** [review]

File: packages/core/useTemporalNow/index.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+   *
+   * @default 'gregory'
+   */
+  calendar?: string
+  /**
+   * Update interval in milliseconds
+   *
+   * @default 1000
+   */
+  interval?: MaybeRefOrGetter<number>
```
Comment (9romise) [inline packages/core/useTemporalNow/index.ts:24]: Maybe we could follow the same approach as `useNow` and inherit from `ConfigurableScheduler`, rather than offering fixed `useIntervalFn` options.

**Review by antfu — APPROVED** [review]
